src/Symfony/Component/DependencyInjection/Loader/YamlFileLoader.php

```diff
@@ -363,6 +363,10 @@ private function parseCallable($callable, $parameter, $id, $file)
                 return array($this->resolveServices($callable[0]), $callable[1]);
             }
 
+            if ('factory' === $parameter && null === $callable[0]) {
```

Comment:
you need to validate ``isset($callable[1])`` though, to accept it only when the second value is not null